### PR TITLE
Add `with*` methods for `WorkflowJob`

### PR DESCRIPTION
**Pull request by zarthross**
- Added `with*` methods for `WorkflowJob` for symmetry with `WorkflowStep`

### CODE REVIEW

**Comment by zarthross** [comment]
Mima is failing because I added methods.  Not sure why since this repo is early-semver...  Should I bump the minor version or add ignores?

**Comment by armanbilge** [comment]
> Not sure why
- https://github.com/lightbend-labs/mima/issues/817

You can add ignores.

File: github-actions/mima.sbt [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import com.typesafe.tools.mima.core._
+mimaBinaryIssueFilters ++= Seq(
```
Comment (armanbilge) [inline github-actions/mima.sbt:2]: I think MiMa might support wildcards, but this is good too.

```scala
ProblemFilters.exclude[ReversedMissingMethodProblem](
  "org.typelevel.sbt.gha.WorkflowJob.with*"
)
```

**Review by armanbilge — APPROVED** [review]
Thank you!

**Pull request merged**